engine/src/main/java/com/datatorrent/stram/engine/Node.java

```diff
@@ -180,12 +182,19 @@ public Operator getOperator()
     return operator;
   }
 
+  public void setRedeploy(boolean redeploy)
+  {
+    this.redeploy = redeploy;
+  }
+
   @Override
   public void setup(OperatorContext context)
   {
     shutdown = false;
     logger.debug("Operator Context = {}", context);
-    operator.setup(context);
+    if (!redeploy) {
+      operator.setup(context);
```

Comment:
As it is a new behaviour, should this be exposed as a new method in the operator interface? This allows us to identify the operators that actually make use of this feature.

Reply:
That is a good idea. I have made some changes for this and will update the PR soon.